src/usr/xio/xio_context.c

```diff
@@ -849,3 +865,177 @@ int xio_ctx_debug_thread_unlock(struct xio_context *ctx)
 
 #endif
 
+void *xio_context_umalloc(struct xio_context *ctx, size_t size)
+{
+	if (ctx && ctx->allocator_assigned &&
+	    ctx->mem_allocator.allocate)
+		return ctx->mem_allocator.allocate(size,
```

Comment:
we need allocator to have ctx in its API as well. this way we will be able to differentiate between different contexts and provide priority in allocations